### PR TITLE
Add GCP vault in 2.8.1.3

### PR DESCRIPTION
**Pull request by lokesh-chechani**
### Summary

Added documentation for GCP Vault to Secret Management section in Kong Gateway 

### Reason
[FTI-4100](https://konghq.atlassian.net/browse/FTI-4100). In discussion with the product engineering team to ship this feature as part of gateway release v2.8.1.3. 

Slack discussion: https://kongstrong.slack.com/archives/C0223LNEQCB/p1658393569090289?thread_ts=1658213496.854579&cid=C0223LNEQCB

### Testing
Minor document update.

### CODE REVIEW

**Review by Guaris — DISMISSED** [review]

**Comment by mayocream** [comment]
Please hold, we are having a discussion about whether or not to put this feature into 2.8.x.

**Comment by Guaris** [comment]
Okay

**Comment by khaikong** [comment]
Can we get this reviewed too @Guaris 
https://github.com/Kong/docs.konghq.com/pull/4165

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  "foo": "bar",
+  "snip": "snap",
+}
+```
+
+You can now reference the secret's individual resources like this: 
+
+```bash
+{vault://gcp/my-secret-name/foo}
+{vault://gcp/my-secret-name/snap}
```
Comment (mayocream) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:41]:
```suggestion
{vault://gcp/my-secret-name/foo?project_id=my_project_id}
{vault://gcp/my-secret-name/snap?project_id=my_project_id}
```

Comment (mayocream) [reply app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:41]: `?project_id` can also be ignored if `GCP_PROJECT_ID` env is set.

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  --data config.project_id="my_project_id"
+```
+
+{% endnavtab %}
+{% navtab HTTPie %}
+
+```bash
+http PUT :8001/vaults-beta/my-gco-sm-vault name="gcp" \
+  description="Storing secrets in GCP Secrets Manager" \
+  config.region="my_project_id" \
```
Comment (mayocream) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:64]:
```suggestion
  config.project_id="my_project_id" \
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+{% endnavtab %}
+{% endnavtabs %}
+
+Result:
+
+```json
+
+{
+    "config": {
+        "project_id": "project-last-hope"
```
Comment (mayocream) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:77]:
```suggestion
        "project_id": "my_project_id"
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    "description": "Storing secrets in GCP Secrets Manager",
+    "id": "90e200be-cf84-4ce9-a1d6-a41c75c79f31",
+    "name": "gcp",
+    "prefix": "my-gcp-sm-vault",
+    "tags": null,
+    "updated_at": 1657874961
+}
+```
+
+With the Vault entity in place, you can now reference the secrets. This allows you to drop the `KONG_VAULT_GCP_PROJECT_ID`
```
Comment (mayocream) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:89]:
```suggestion
With the Vault entity in place, you can now reference the secrets. This allows you to drop the `GCP_PROJECT_ID`
```

**Comment by cloudjumpercat** [comment]
Can someone tag me when all your changes are complete so I can review them? Thanks!

**Comment by hanshuebner** [comment]
@kikito Please tag Diana when these changes are complete.

**Comment by kikito** [comment]
I had to rebase this branch over `main`, otherwise reviews were too confusing. If you had a local copy, you might have to delete your local branch (`git checkout main && git branch -D feat/fti_4100_gcp_vault`) and re-clone (`git fetch --all && git switch feat/fti_4100_gcp_vault )

**Review by kikito — APPROVED** [review]

**Comment by kikito** [comment]
I think the docs are good enough now, I have approved the PR. Tagging @cloudjumpercat as requested.

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/advanced-usage.md [hunk 5]
```diff
@@ -75,10 +68,7 @@ Options:
 ## Vaults Entity
 
 {:.warning}
-> **Beta warning:**
-> <br>
-> The API endpoint is suffixed with `-beta` to avoid any possible conflicts. This will be
-> changed in the future. Kong Manager has currently no supports for configuring vault entities.
+> Kong Manager has currently no supports for configuring vault entities.
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/advanced-usage.md:71]:
```suggestion
> Kong Manager currently doesn't support configuring vault entities.
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/advanced-usage.md [hunk 6]
```diff
@@ -88,7 +78,7 @@ Create a Vault entity:
 {% navtab cURL %}
 
 ```bash
-$ curl -i -X PUT http://<hostname>:8001/vaults-beta/my-env-vault-1  \
+$ curl -i -X PUT http://<hostname>:8001/vaults/my-env-vault-1  \
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/advanced-usage.md:81]:
```suggestion
$ curl -i -X PUT http://HOSTNAME:8001/vaults/my-env-vault-1  \
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/aws-sm.md [hunk 7]
```diff
@@ -43,7 +42,7 @@ The Vault entity can only be used once the database is initialized. Secrets for
 {% navtab cURL %}
 
 ```bash
-curl -i -X PUT http://<hostname>:8001/vaults-beta/my-aws-sm-vault  \
+curl -i -X PUT http://<hostname>:8001/vaults/my-aws-sm-vault  \
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/aws-sm.md:45]:
```suggestion
curl -i -X PUT http://HOSTNAME:8001/vaults/my-aws-sm-vault  \
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/env.md [hunk 8]
```diff
@@ -46,7 +45,7 @@ The Vault entity can only be used once the database is initialized. Secrets for
 {% navtab cURL %}
 
 ```bash
-curl -i -X PUT http://<hostname>:8001/vaults-beta/my-env-vault \
+curl -i -X PUT http://<hostname>:8001/vaults/my-env-vault \
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/env.md:48]:
```suggestion
curl -i -X PUT http://HOSTNAME:8001/vaults/my-env-vault \
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+To configure GCP secrets manager, the `GCP_SERVICE_ACCOUNT`
+environment variable must be set to the JSON document referring to the
+[credentials for your service account](https://cloud.google.com/iam/docs/creating-managing-service-account-keys):
+
+```bash
+export GCP_SERVICE_ACCOUNT=$(cat gcp-my-project-c61f2411f321.json)
+```
+
+{{site.base_gateway}} will use the key to automatically authenticate
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:24]:
```suggestion
{{site.base_gateway}} uses the key to automatically authenticate
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+with the GCP API and grant you access.
+
+To use GCP Secrets Manager with
+[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)
+on a GKE cluster, update your pod spec so that the service account is
+attached to the pod. For configuration information, read the Workload
+Identity configuration
+[documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to).
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:32]:
```suggestion
attached to the pod. For configuration information, read the [Workload
Identity configuration
documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to).
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+{:.note}
+> With Workload Identity, setting the `GCP_SERVICE_ACCOUNT` is not necessary.
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:35]:
```suggestion
> With Workload Identity, setting the `GCP_SERVICE_ACCOUNT` isn't necessary.
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+The current version of {{site.base_gateway}}'s implementation supports
+configuring
+[GCP Secrets Manager](https://cloud.google.com/secret-manager/) in two
+ways:
+
+* Environment variables
+* Workload Identity
+
+To configure GCP secrets manager, the `GCP_SERVICE_ACCOUNT`
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:16]:
```suggestion
To configure GCP Secrets Manager, the `GCP_SERVICE_ACCOUNT`
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+{vault://gcp/my-secret-name/foo?project_id=my_project_id}
+{vault://gcp/my-secret-name/snip?project_id=my_project_id}
+```
+
+Note that both the provider (`gcp`) as well as the GCP project ID
+(`my_project_id`) need to be specified.
+
+## Entity
+
+Once the database has been initialized, a Vault entity can be created
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:63]:
```suggestion
Once the database is initialized, a Vault entity can be created
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+that encapsulates the provider and the GCP project ID:
+
+{% navtabs codeblock %}
+{% navtab cURL %}
+
+```bash
+curl -i -X PUT http://<hostname>:8001/vaults/my-gcp-sm-vault \
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:70]:
```suggestion
curl -i -X PUT http://HOSTNAME:8001/vaults/my-gcp-sm-vault \
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+  --data name=gcp \
+  --data description="Storing secrets in GCP Secrets Manager" \
+  --data config.project_id="my_project_id"
+```
+
+{% endnavtab %}
+{% navtab HTTPie %}
+
+```bash
+http -f PUT http://<hostname>:8001/vaults/my-gcp-sm-vault \
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:80]:
```suggestion
http -f PUT http://HOSTNAME:8001/vaults/my-gcp-sm-vault \
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+With the Vault entity in place, you can reference the GCP secrets
+through it:
+
+```bash
+{vault://my-gcp-sm-vault/my-secret-name/foo}
+{vault://my-gcp-sm-vault/my-secret-name/snip}
+```
+
+When using the Vault entity, the GCP project ID no longer needs to be
+specified to access the secrets.
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/gcp-sm.md:116]:
```suggestion
When you use the Vault entity, you no longer need to specify the GCP project ID to access the secrets.
```

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/hashicorp-vault.md [hunk 17]
```diff
@@ -29,7 +29,7 @@ The Vault entity can only be used once the database is initialized. Secrets for
 {% navtab cURL %}
 
 ```bash
-curl -i -X PUT http://<hostname>:8001/vaults-beta/my-hashicorp-vault \
+curl -i -X PUT http://<hostname>:8001/vaults/my-hashicorp-vault \
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/hashicorp-vault.md:32]:
```suggestion
curl -i -X PUT http://HOSTNAME:8001/vaults/my-hashicorp-vault \
```

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]

**Review by cloudjumpercat — APPROVED** [review]

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/index.md [hunk 18]
```diff
@@ -51,29 +50,28 @@ documentation for each plugin to identify the referenceable fields:
...
-* In this version, this feature isn't enabled by default. To test it out, start
-{{site.base_gateway}} with `KONG_VAULTS=bundled` if running Kong in a container,
-or with `vaults=bundled` set in `kong.conf`.
-* The API endpoint is suffixed with `-beta` to avoid any possible conflicts. This
-endpoint will change once the beta is over.
+In this version, this feature isn't enabled by default due to
+conflicts with previous releases of {{site.base_gateway}}.  To enable
+it, start {{site.base_gateway}} with `KONG_VAULTS=bundled
+KONG_VAULTS_USE_NEW_STYLE_API=on` if running Kong in a container, or with
+`vaults=bundled` and `vaults_use_new_style_api=on`set in `kong.conf`.
```
Comment (mayocream) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/index.md:70]:
```suggestion
In the GA release of the Secrets Management feature in
{{site.base_gateway}}, this feature is enabled by default.
Due to conflicts with previous releases of {{site.base_gateway}},
the endpoints for secrets management in the
Admin API will be moved from the previous `/vaults-beta` prefix to
`/vaults` with `vaults_use_new_style_api=on` set in `kong.conf`.
```

**Review by mayocream — APPROVED** [review]

**Comment by mayocream** [comment]
@cloudjumpercat Please take another look, and the autodoc Admin API and configuration need re-generation when 2.8.1.3 is released.

File: app/_hub/kong-inc/vault-auth/0.4.0.md [hunk 19]
```diff
@@ -0,0 +1,291 @@
...
+      description: |
+        An optional boolean value telling the plugin to show or hide the credential from the upstream service. If `true`, the plugin will strip the credential from the request (i.e. the header or querystring containing the key) before proxying it.
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string
+      description: |
+        An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`.
+
+        **Note:** This value must refer to the Consumer `id` attribute that is internal to {{site.base_gateway}}, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/vault-auth/0.4.0.md:64]:
```suggestion
        **Note:** This value must refer to the Consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/vault-auth/_index.md [hunk 20]
```diff
@@ -61,7 +61,7 @@ params:
       description: |
         An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`.
 
-        **Note:** This value must refer to the Consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
+        **Note:** This value must refer to the Consumer `id` attribute that is internal to {{site.base_gateway}}, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/vault-auth/_index.md:64]:
```suggestion
        **Note:** This value must refer to the Consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]

File: app/_hub/kong-inc/vault-auth/0.4.0.md (hunk 19 above)
Comment (cloudjumpercat) [inline app/_hub/kong-inc/vault-auth/0.4.0.md:64]:
```suggestion
        **Note:** This value must refer to the Consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

**Comment by tyler-ball** [comment]
I added https://github.com/Kong/docs.konghq.com/pull/4144/commits/63f59e38496bb6f78d380f30094b10fb946918ab - @cloudjumpercat @mayocream can y'all review that?

File: app/gateway/2.8.x/plan-and-deploy/security/secrets-management/getting-started.md [hunk 21]
```diff
@@ -1,21 +1,19 @@
...
 ## Getting started
 
 {:.note}
-> This feature isn't enabled by default in this version of Kong.
+> This feature isn't enabled by default in this version of {{site.base_gateway}}.
+><br>
+> Start {{site.base_gateway}} with `KONG_VAULTS=bundled KONG_VAULTS_USE_NEW_STYLE_API=on`.
 ><br>
-> Start Kong Gateway with `KONG_VAULTS=bundled`.
+> When running in Hybrid or DBLess mode, Secrets Management is only supported in {{site.base_gateway}} version 2.8.1.3 and greater.
```
Comment (cloudjumpercat) [inline app/gateway/2.8.x/plan-and-deploy/security/secrets-management/getting-started.md:16]:
```suggestion
> When running {{site.base_gateway}} in hybrid or DB-less mode, secrets management is only supported in {{site.base_gateway}} 2.8.1.3 or later.
```

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]

**Review by cloudjumpercat — APPROVED** [review]

**Pull request merged**